tests/hf_generation_test.py

```
@@ -0,0 +1,160 @@
+model = HFGPTBaseForCausalLM(get_m2rnn_moe_test_config(num_layers=2))
+model.train()
+input_ids = torch.randint(3, model.config.vocab_size, (2, 16))
+loss = model(input_ids=input_ids, labels=input_ids).loss
+assert torch.isfinite(loss), f"non-finite loss {loss}"
+"""
+
+    repo_root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    env = os.environ.copy()
+    env.update({"MASTER_ADDR": "localhost", "MASTER_PORT": "29513", "WORLD_SIZE": "1", "RANK": "0"})
+    env["PYTHONPATH"] = repo_root + os.pathsep + env.get("PYTHONPATH", "")
+    result = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)